src/lib/comms/sol-lwm2m-security.c

```diff
@@ -255,6 +263,328 @@ get_id_from_client(const void *data, struct sol_network_link_addr *addr, char *i
+        PUBLIC_KEY_Y, SECURITY_PUBLIC_KEY_OR_IDENTITY);
+}
+
+static int
+verify_ecdsa_key_from_server_or_bs_server(const void *data,
+    struct sol_network_link_addr *addr, const unsigned char *other_pub_x,
+    const unsigned char *other_pub_y, size_t key_size)
+{
+    unsigned char buf_aux[SOL_DTLS_ECDSA_PUB_KEY_X_LEN];
+    int r;
+
+    r = get_ecdsa_field_from_res_id(data, addr, buf_aux,
+        SERVER_PUBLIC_KEY_X, SECURITY_SERVER_PUBLIC_KEY);
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (!memcmp(other_pub_x, buf_aux, SOL_DTLS_ECDSA_PUB_KEY_X_LEN))
+        SOL_DBG("Server's Public Key (x coord.) matches");
+    else {
+        SOL_WRN("Server's Public Key (x coord.) does not match");
+        return -EINVAL;
+    }
+
+    r = get_ecdsa_field_from_res_id(data, addr, buf_aux,
+        SERVER_PUBLIC_KEY_Y, SECURITY_SERVER_PUBLIC_KEY);
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (!memcmp(other_pub_y, buf_aux, SOL_DTLS_ECDSA_PUB_KEY_Y_LEN))
+        SOL_DBG("Server's Public Key (y coord.) matches");
+    else {
+        SOL_WRN("Server's Public Key (y coord.) does not match");
+        return -EINVAL;
+    }
+
+    return 0;
+}
+
+static int
+verify_ecdsa_key_from_client(const void *data,
+    struct sol_network_link_addr *addr, const unsigned char *other_pub_x,
+    const unsigned char *other_pub_y, size_t key_size)
+{
+    const struct sol_lwm2m_security *ctx = data;
+    struct sol_lwm2m_server *lwm2m_server;
+    struct sol_lwm2m_bootstrap_server *lwm2m_bs_server;
+    struct sol_ptr_vector *known_pub_keys;
+    struct sol_blob *cli_pub_key;
+    uint16_t i;
+
+    if (ctx->type == LWM2M_SERVER) {
+        SOL_DBG("ctx->type = LWM2M_SERVER");
+        lwm2m_server = ctx->entity;
+        known_pub_keys = &lwm2m_server->known_pub_keys;
+    } else {
+        SOL_DBG("ctx->type = LWM2M_BOOTSTRAP_SERVER");
+        lwm2m_bs_server = ctx->entity;
+        known_pub_keys = &lwm2m_bs_server->known_pub_keys;
+    }
+
+    SOL_PTR_VECTOR_FOREACH_IDX (known_pub_keys, cli_pub_key, i) {
+        if (cli_pub_key->size !=
+            (SOL_DTLS_ECDSA_PUB_KEY_X_LEN + SOL_DTLS_ECDSA_PUB_KEY_Y_LEN)) {
+            SOL_WRN("The stored Client's Public Key '%.*s' is %zu-bytes long;"
+                " expecting a %d-bytes long Public Key",
+                SOL_STR_SLICE_PRINT(sol_str_slice_from_blob(cli_pub_key)),
+                cli_pub_key->size,
+                SOL_DTLS_ECDSA_PUB_KEY_X_LEN + SOL_DTLS_ECDSA_PUB_KEY_Y_LEN);
+            return -EINVAL;
+        }
+
+        if (!memcmp(other_pub_x, cli_pub_key->mem, SOL_DTLS_ECDSA_PUB_KEY_X_LEN)) {
+            SOL_DBG("Stored Client's Public Key (x coord.) matches");
+            if (!memcmp(other_pub_y,
+                (unsigned char *)cli_pub_key->mem + SOL_DTLS_ECDSA_PUB_KEY_X_LEN,
+                SOL_DTLS_ECDSA_PUB_KEY_Y_LEN)) {
+                SOL_DBG("Stored Client's Public Key (y coord.) matches");
+                return 0;
+            }
+        }
```

Comment:
I can't do it for the pub_x part, since if it doesn't match it means I just need to keep going on the loop till I find a matching key. I can do it for the Y part, though, since it can even be computed by the X part: http://bitcoin.stackexchange.com/questions/19666/why-do-keys-need-both-x-and-y-coordinates-if-x-can-be-solved-for-y-using-the-cu